### ISSUE TITLE
reactions problem getting single post

### ISSUE THREAD

**Issue by emmanuelasso** [kevinzg/facebook-scraper#688]
I have a curious problem. According to another issue, a solution when you stop getting reactions from a post is to run `pip uninstall facebook-scraper` twice before `pip install`

After I do this, the reactions are fetched, but the second time I run the `get_post` function passing a url (see code below) it stops fetching the reactions. If I uninstall the library and reinstall it it works again but only for the first query.

```
posts = list(get_posts(
                        post_urls=["https://www.facebook.com/memesdecolombiaaa/posts/999033537657270"],cookies=cookies_path,
                        options={"comments" : "generator", "reactions" : True, "allow_extra_requests": False}
))
end = time.time()
pprint.pprint(posts)
```

The output I get on the first run results in:

```
[{'available': True,
  'comments': 154,
  'comments_full': <generator object PostExtractor.extract_comments_full at 0x7fa6e9551890>,
  'factcheck': None,
  'fetched_time': datetime.datetime(2022, 2, 24, 22, 55, 36, 54347),
  'image': None,
  .....
  'is_live': False,
  'likes': 200,
  'link': 'https://www.instagram.com/memecolombianos_/?fbclid=IwAR3Qfbv2Aj4uAajp9b77MPwUI3JSb3nktu_bTLGP2qnQhYKoSMPIzJ1xcl4',
  'links': [],
  'original_request_url': 'https://www.facebook.com/memesdecolombiaaa/posts/999033537657270',
  'original_text': None,
  'page_id': '322611095299521',
  'post_id': 999033537657270,
  'post_text': 'JJAJAJAJAJJAJA',
  'post_url': 'https://facebook.com/story.php?story_fbid=999033537657270&id=322611095299521',
  'reaction_count': 1355,
  'reactions': {'care': 2,
                'haha': 1147,
                'like': 200,
                'love': 3,
                'sad': 1,
                'wow': 2},
  'reactors': [],
  'shared_post_id': None,
  'shared_post_url': None,
  'shared_text': '',
  'shared_time': None,
  'shared_user_id': None,
  'shared_username': None,
  'sharers': None,
  'shares': 852,
  'text': 'JJAJAJAJAJJAJA',
  'time': datetime.datetime(2022, 2, 24, 19, 55, 36),
  'timestamp': None,
  'user_id': '322611095299521',
  'user_url': 'https://facebook.com/memesdecolombiaaa/?__tn__=C-R',
  'username': 'Memes Colombianos',
  'video': None,
  'video_duration_seconds': None,
  'video_height': None,
  'video_id': None,
  'video_quality': None,
  'video_size_MB': None,
  'video_thumbnail': None,
  'video_watches': None,
  'video_width': None,
  'w3_fb_url': 'https://www.facebook.com/story.php?story_fbid=999033537657270&id=322611095299521',
  'was_live': False,
  'with': None}]
```

But from the second time I run the function the output doesn't have the reactions:

```
[{'available': True,
  'comments': 0,
  'comments_full': <generator object PostExtractor.extract_comments_full at 0x7fb168b82f20>,
  'factcheck': None,
  'image': None,
...
  'is_live': False,
  'likes': 1300,
  'link': 'https://www.instagram.com/memecolombianos_/?fbclid=IwAR1yRtWNu6qPSzyrko6CS-N0Co_RECAHhXR30yeSo_gTNqcuZkRLrsoU-\\-\\k',
  'links': [],
  'original_request_url': 'https://www.facebook.com/memesdecolombiaaa/posts/999033537657270',
  'original_text': None,
  'page_id': '322611095299521',
  'post_id': '999033537657270_999067444320546',
  'post_text': 'JJAJAJAJAJJAJA',
  'post_url': 'https://facebook.com/story.php?story_fbid=999033537657270&id=322611095299521',
  'reaction_count': None,
  'reactions': None,
  'reactors': None,
  'shared_post_id': None,
  'shared_post_url': None,
  'shared_text': '',
  'shared_time': None,
  'shared_user_id': None,
  'shared_username': None,
  'sharers': None,
  'shares': 852,
  'text': 'JJAJAJAJAJJAJA',
  'time': datetime.datetime(2022, 2, 24, 19, 55, 40),
  'timestamp': None,
  'user_id': '322611095299521',
  'user_url': 'https://facebook.com/memesdecolombiaaa/?__tn__=C-R',
  'username': 'Memes Colombianos',
  'video': None,
  'video_duration_seconds': None,
  'video_height': None,
  'video_id': None,
  'video_quality': None,
  'video_size_MB': None,
  'video_thumbnail': None,
  'video_watches': None,
  'video_width': None,
  'w3_fb_url': None,
  'was_live': False,
  'with': None}]

```

**Comment by neon-ninja** [comment]
Reaction extraction works for me for this post, repeatedly. Can you please try enable debug logs `enable_logging()`, and post the logs you get when reaction extraction fails?

**Comment by Stoakes** [comment]
Hello,
I got a similar issue (reactions not fetched), so I ran the example above and indeed got no reactions:

```
>>> for post in get_posts(post_urls=['https://www.facebook.com/memesdecolombiaaa/posts/999033537657270"'], options={"comments" : "generator", "reactions" : True, "allow_extra_requests": False}):
...      pprint.pprint(post)
... 
Requesting page from: https://m.facebook.com/memesdecolombiaaa/posts/999033537657270"
Got exact timestamp from publish_time: 2022-02-25 01:50:14
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_photo_link didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_video didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_video_thumbnail didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_video_id didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_video_meta didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_factcheck didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_share_information didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_listing didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Extract method extract_with didn't return anything
[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Exception while extracting reactions: KeyError('reactors')
{'available': True,
 'comments': 590,
 'comments_full': <generator object PostExtractor.extract_comments_full at 0x7f79713cbc10>,
 'factcheck': None,
 'image': None,
 'image_id': '999033517657272',
 'image_ids': ['999033517657272'],
 'image_lowquality': 'https://scontent-cdg2-1.xx.fbcdn.net/v/t39.30808-6/274748322_999033607657263_1903024647675946317_n.jpg?stp=cp0_dst-jpg_e15_p320x320_q65&_nc_cat=108&ccb=1-5&_nc_sid=110474&_nc_ohc=q6vtSDjJZTAAX_-l3DF&_nc_ht=scontent-cdg2-1.xx&oh=00_AT8vIoWtY7j6mlfw5QWiqyXkBIlFG7NfFeYXWWMosNeQUQ&oe=6275237C',
 'images': None,
 'images_description': None,
 'images_lowquality': ['https://scontent-cdg2-1.xx.fbcdn.net/v/t39.30808-6/274748322_999033607657263_1903024647675946317_n.jpg?stp=cp0_dst-jpg_e15_p320x320_q65&_nc_cat=108&ccb=1-5&_nc_sid=110474&_nc_ohc=q6vtSDjJZTAAX_-l3DF&_nc_ht=scontent-cdg2-1.xx&oh=00_AT8vIoWtY7j6mlfw5QWiqyXkBIlFG7NfFeYXWWMosNeQUQ&oe=6275237C'],
 'images_lowquality_description': ['May be a meme of 4 people, outdoors and '
                                   "text that says 'Durante el camino Foto en "
                                   "redes'"],
 'is_live': False,
 'likes': 616,
 'link': 'https://www.instagram.com/memecolombianos_/',
 'links': [],
 'original_request_url': 'https://www.facebook.com/memesdecolombiaaa/posts/999033537657270"',
 'original_text': None,
 'page_id': '322611095299521',
 'post_id': 'pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl',
 'post_text': 'JJAJAJAJAJJAJA',
 'post_url': 'https://facebook.com/story.php?story_fbid=pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl&id=322611095299521',
 'reaction_count': None,
 'reactions': None,
 'reactors': None,
 'shared_post_id': None,
 'shared_post_url': None,
 'shared_text': '',
 'shared_time': None,
 'shared_user_id': None,
 'shared_username': None,
 'sharers': None,
 'shares': 3793,
 'text': 'JJAJAJAJAJJAJA',
 'time': datetime.datetime(2022, 2, 25, 1, 50, 14),
 'timestamp': 1645750214,
 'user_id': '322611095299521',
 'user_url': 'https://facebook.com/memesdecolombiaaa/?locale2=en_US&__tn__=C-R',
 'username': 'Memes Colombianos',
 'video': None,
 'video_duration_seconds': None,
 'video_height': None,
 'video_id': None,
 'video_quality': None,
 'video_size_MB': None,
 'video_thumbnail': None,
 'video_watches': None,
 'video_width': None,
 'w3_fb_url': None,
 'was_live': False,
 'with': None}

```

In a word, I can see the error `[pfbid0VicQZcytiqjwHrjcPjGZz2t59mcY3UCc4bpNZCd4YNvVmgdsQv1cTWVkDbQDbCuJl] Exception while extracting reactions: KeyError('reactors')`

The issue is not limited to this page.

Here is the version I'm running:
```bash
pip3 show facebook-scraper
Name: facebook-scraper
Version: 0.2.55
Summary: Scrape Facebook public pages without an API key
Home-page: https://github.com/kevinzg/facebook-scraper
Author: Kevin Zúñiga
Author-email: kevin.zun@gmail.com
License: MIT
Location: /home/stoakes/.local/lib/python3.8/site-packages
Requires: dateparser, demjson3, requests-html
Required-by: 

```

**Comment by Stoakes** [comment]
In case your are reading this issue because you don't have reactions in response, make sure you've set cookies correctly.

More details at https://github.com/kevinzg/facebook-scraper/discussions/743 

Btw, this issue can probably be closed.

**Comment by neon-ninja** [comment]
@Stoakes it's up to @emmanuelasso to close this issue when it's solved for them
